File: java/spring-data/spring-data-jdbc-querydsl/src/main/java/sample/data/Product.java
package sample.data;

import java.math.BigDecimal;
import java.sql.Timestamp;

public class Product {
    private long id;
    private String name;
    private BigDecimal price;
    private Timestamp releaseDate = new Timestamp(System.currentTimeMillis());

    public long getId() {
        return id;
    }
    public void setId(long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }
    public void setName(String name) {
        this.name = name;
    }

    public BigDecimal getPrice() {
        return price;
    }
    public void setPrice(BigDecimal price) {
        this.price = price;
    }

    public Timestamp getReleaseDate() {
        return releaseDate;
    }
    public void setReleaseDate(Timestamp releaseDate) {
        this.releaseDate = releaseDate;
    }
}

File: scala/querydsl/sample-beans/src/main/scala/sample/model/Product.scala
package sample.model


object Product extends QProduct("product") {
  override def as(variable: String) = new QProduct(variable)
  
}

/**
 * Product is a Querydsl bean type
 */
class Product {

  var id: java.lang.Long = _

  var name: String = _

  var price: java.lang.Long = _

  var releaseDate: java.sql.Timestamp = _

}

